src/CreateBoardForm.tsx

```diff
@@ -2,51 +2,30 @@ import React, { Component, FormEvent } from 'react'
 import { remove, shuffle } from 'lodash'
 import { Board } from './Board'
 import { Checkbox } from './Checkbox'
+import { database } from './firebaseConfig'
 
-export class CreateBoardForm extends Component<{ togglePage: any }, { chosenCharacters: string[], showBoard: boolean, board: string[], boardName: string }> {
-  baseCharacters: string[] = [
-    "Arya Stark",
-    "Sansa Stark",
-    "Bran Stark",
-    "Jon Snow",
-    "Theon Greyjoy",
-    "Yara Greyjoy",
-    "Euron Greyjoy",
-    "Cersei Lannister",
-    "Jaime Lannister",
-    "Tyrion Lannister",
-    "Sam Tarly",
-    "Tormund Giantsbane",
-    "Brienne of Tarth",
-    "Gendry",
-    "Podrick",
-    "Danaerys Targaryen",
-    "Jorah Mormont",
-    "Davos Seaworth",
-    "Lyanna Mormont",
-    "The Night King"
-  ]
+export class CreateBoardForm extends Component<{ togglePage: any }, { chosenCharacters: string[], showBoard: boolean, board: string[], boardName: string, baseCharacters: string[], optionalCharacters: string[] }> {
+  constructor(props) {
+    super(props)
+    this.state = {
+      chosenCharacters: [] as string[],
+      showBoard: false,
+      board: [] as string[],
+      boardName: '',
+      baseCharacters: [] as string[],
+      optionalCharacters: [] as string[]
+    }
+
+    this.getCharacters()
+  }
 
-  optionalCharacters: string[] = [
-    "Lord Varys",
-    "Missandei",
-    "Grey Worm",
-    "The Hound",
-    "Beric Dondarrion",
-    "Bronn",
-    "Melisandre",
-    "Edd Tollett",
-    "Gilly",
-    "Little Sam",
-    "Drogon",
-    "Rhaegal"
-  ]
+  getCharacters = (): void => {
+    database.ref('/characters').once('value').then(characters => {
+      const baseCharacters: string[] = characters.val().filter(character => character.required).map(character => character.name)
+      const optionalCharacters: string[] = characters.val().filter(character => !character.required).map(character => character.name)
 
-  state = {
-    chosenCharacters: [] as string[],
-    showBoard: false,
-    board: [] as string[],
-    boardName: ''
+      this.setState({ baseCharacters, optionalCharacters })
```

Comment:
nit pick, probably should call them `baseCharacterNames` etc b/c they're no longer "characters" the object type.